packages/lean_analytics/lib/core/event.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/foundation.dart';
+
+class AnalyticsEvent {
+  AnalyticsEvent({
+    required this.name,
+    this.params,
+  });
+
+  final Map<String, Object>? params;
+  final String name;
+}
+
+enum ScreenViewType {
+  push,
+  pop,
+  restore;
+}
+
+class ScreenViewAnalyticsEvent extends AnalyticsEvent {
+  ScreenViewAnalyticsEvent({
+    required String pageName,
+    required ScreenViewType screenViewType,
+    Map<String, Object>? params,
+  }) : super(
+          name: pageName,
+          params: {
+            if (params != null) ...params,
+            'page_name': pageName,
+            'action_type': screenViewType.name,
```

Comment:
Do we want to overwrite user-provided params `page_name`/`action_type`? If so, maybe we should error out if a user provides them since its a noop. Otherwise it will be a silent bug

Reply:
Ok, I add asserts